Neo4j.Driver/Neo4j.Driver/Internal/ConnectionPool.cs

```diff
@@ -151,6 +149,17 @@ public IPooledConnection CreateNewPooledConnection()
         private PooledConnection NewPooledConnection()
         {
             Interlocked.Increment(ref _poolSize);
+            if (_maxPoolSize != Config.Infinite && _poolSize > _maxPoolSize)
```

Comment:
If we have two threads and `_poolSize == _maxPoolSize  - 1` they both will increment `_poolSize` and throw, right? I think one of them should succeed. We could probably use result of `Interlocked.Increment` to figure out which one should fail.